### PR TITLE
#10  : Add payload validator

### PR DESCRIPTION
**Pull request by raghavdhupar7**
#10  :Added  params validation to GET@/users/:userId, PUT@/users/:userId, and DELETE@/users/:userId
Added body validation to PUT@/users/:userId, and POST@/users/:userId

### CODE REVIEW

File: routes/users.js [hunk 1]
```diff
@@ -2,7 +2,29 @@ const Users = require("../models/user");
...
+            type: "string",
+            pattern: /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/,
+        },
+        gender: { type: "string", values: ["male", "female", "other"] },
+        username: { type: "string", pattern: /^[a-zA-Z0-9]+$/ },
+        birthday: { type: "number", integer: true },
+    },
+};
+
+router.get("/", RequestValidator(userSchema), async (req, res, next) => {
```
Comment (tanayhub) [inline routes/users.js:27]: why would `GET` requests need params/body validators?

File: routes/users.js [hunk 2]
```diff
@@ -11,7 +33,7 @@ router.get("/", async (req, res, next) => {
     }
 });
 
-router.get("/:userId", async (req, res, next) => {
+router.get("/:userId", RequestValidator(userSchema), async (req, res, next) => {
```
Comment (tanayhub) [inline routes/users.js:36]: same

File: routes/users.js [hunk 3]
```diff
@@ -31,31 +53,37 @@ router.post("/", async (req, res, next) => {
...
 });
 
-router.delete("/:userId", async (req, res, next) => {
-    const { userId } = req.params;
-    try {
-        const user = await Users.findByIdAndRemove(userId);
-        return user ? res.status(200).end() : res.status(404).end();
-    } catch (error) {
-        next(error);
+router.delete(
```
Comment (tanayhub) [inline routes/users.js:72]: same

File: routes/users.js [hunk 4]
```diff
@@ -31,31 +53,37 @@ router.post("/", async (req, res, next) => {
...
+            return user ? res.status(200).end() : res.status(404).end();
+        } catch (error) {
+            next(error);
+        }
     }
-});
+);
 
-router.delete("/", async (req, res, next) => {
+router.delete("/", RequestValidator(userSchema), async (req, res, next) => {
```
Comment (tanayhub) [inline routes/users.js:86]: same

**Review by tanayhub — CHANGES_REQUESTED** [review]

File: routes/users.js (hunk 1 above)
Comment (raghavdhupar7) [reply routes/users.js:27]: was mentioned in the description of the story

**Review by tanayhub — APPROVED** [review]

**Pull request merged**